userspace/libsinsp/gen_filter.cpp

```diff
@@ -427,6 +439,11 @@ std::string gen_event_filter_factory::filter_fieldclass_info::as_string(bool ver
 		os << std::endl;
 	}
 
+	if(deprecated_count == fields.size())
```

Comment:
Why do we return an empty string here? 
We don't do that for `skippable` ones :/

Also, do you think we could merge the `is_deprecated` into `is_skippable` (ie by adding an additional check on `EPF_DEPRECATED`)?
I am worried that we might use the same `is_skippable` method in other part of the code, and managing the new flag in the old method would allow us to skip deprecated fields everywhere for free.

Reply:
The idea behind this is that if all the fields are deprecated, the field class is deprecated as well: in that case we don't want to print out the field class at all.

I think could be a good idea but I have a concern: if all fields are EPF_TABLE_ONLY, should we omit the field class from the `--list`?

Reply:
> The idea behind this is that if all the fields are deprecated, the field class is deprecated as well: in that case we don't want to print out the field class at all.

Makes sense!

> I think could be a good idea but I have a concern: if all fields are EPF_TABLE_ONLY, should we omit the field class from the --list?

I am not sure :/ 
I checked and `is_skippable` is used only twice; i think we can manage to keep the new `is_deprecated` method so that it is much cleaner too.

Also, should we update `as_markdown` method too?

Reply:
Sure! I totally forgot about that! 😄